### PR TITLE
Fix CFS warnings

### PR DESCRIPTION
**Pull request by csujedihy**
## Description

Add NuGet.config and cargo config file to use Azure artifacts so we can avoid CFS warnings in OneBranch pipeline. We don't actually use them in the pipeline but their presence makes the warnings go away.

## Testing

OB CI

### CODE REVIEW

**Comment by nibanks** [comment]
> We don't actually use them in the pipeline but their presence makes the warnings go away.

Looks like _something_ is trying to use them, especially the Nuget config. We have an exception for the Nuget already I believe. Maybe just try to fix/ignore the Cargo one for now?

**Comment by csujedihy** [comment]
> > We don't actually use them in the pipeline but their presence makes the warnings go away.
> 
> Looks like _something_ is trying to use them, especially the Nuget config. We have an exception for the Nuget already I believe. Maybe just try to fix/ignore the Cargo one for now?

Wow, I added 2 new files and existing tests broke. I didn't expect that. These configs only make sense for OneBranch pipelines. I wonder if I can make the github workflow not pick them up.

**Comment by nibanks** [comment]
> Another proper way to fix it would be actually authenticate with the sources defined in the configs.

What about the exception route? That worked for Nuget.

**Comment by csujedihy** [comment]
> > Another proper way to fix it would be actually authenticate with the sources defined in the configs.
> 
> What about the exception route? That worked for Nuget.

I took another look at the warnings. They were emitted due to Cargo config. Though, the pipeline output still flagged nuget config but they didn't end up being actual warnings.

So, I guess we only need to deal with cargo config.

File: .github/workflows/cargo.yml [hunk 1]
```diff
@@ -40,6 +40,12 @@ jobs:
     - name: Install Cargo
       if: runner.os == 'Linux'
       run: curl https://sh.rustup.rs -sSf | sh -s -- -y
+    # The cargo config file is required for OneBranch CFS.
+    # Remove it for this workflow. Otherwise, cargo will try to reach the internal feed
```
Comment (nibanks) [inline .github/workflows/cargo.yml:44]: There's got to be a better way...

Comment (csujedihy) [reply .github/workflows/cargo.yml:44]: I have seen some proposals here https://internals.rust-lang.org/t/problems-of-cargo-config-files-and-possible-solutions/12987/8 but I have not seen an elegant way to ignore .cargo/config.toml. We could also move the cargo file into a child folder, where cargo would try to locate the config file there.e

Comment (nibanks) [reply .github/workflows/cargo.yml:44]: I'm worried while this fixed **our** repo, other consumers of our crate will be broken by this. Is there no way to tell OneBranch to ignore this?

Comment (csujedihy) [reply .github/workflows/cargo.yml:44]: What do you mean by other consumers of our crate will be broken by this?

Comment (csujedihy) [reply .github/workflows/cargo.yml:44]: We delete this config file so our crate behaves the same as it originally did. What will be broken?

Comment (csujedihy) [reply .github/workflows/cargo.yml:44]: There is a way to tell CFS to ignore everything at repo level by submitting a form.

Comment (nibanks) [reply .github/workflows/cargo.yml:44]: Let's go that route.

**Pull request closed without merging**